src/pages/search.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import axios from 'axios';
+import { RouteChildrenProps } from 'react-router';
+import { Hero } from '../components/hero/hero';
+import { SearchBar } from '../components/searchBar/searchBar';
+import { SearchLogo } from '../components/searchLogo/searchLogo';
+import { ICharacter } from '../intarface/interface';
+
+interface ISearchState {
+    characters: ICharacter[];
+    searchQuery: string;
+}
+
+
+export class Search extends React.Component<RouteChildrenProps, ISearchState> {
+    constructor(props: RouteChildrenProps) {
+        super(props);
+        const searchParam = new URLSearchParams(this.props.location.search).get("search");
+        this.state = {characters: [], searchQuery: searchParam ?? ""};
+      }
+
+      getComics() {
+        axios.get("http://gateway.marvel.com/v1/public/characters", {
+            params: {
+                apikey: "cffc6e60c4ebf25fdb0c2a3a87cfdc6f",
```

Comment:
store api keys in .env variables, it's not secure to write them explicitly in code